.github/workflows/continuous-integration-osx.yml

```diff
@@ -12,19 +12,39 @@ permissions: read-all
 jobs:
   osxci:
     name: osx-ci
-    runs-on: [macos-latest]
 
+    continue-on-error: true
     strategy:
       matrix:
         include:
           - backend: "SERIAL"
             cmake_build_type: "RelWithDebInfo"
+            cxx_standard: "17"
+            compiler: "c++"
+            runner: "macos-latest"
           - backend: "THREADS"
             cmake_build_type: "RelWithDebInfo"
+            cxx_standard: "17"
+            compiler: "c++"
+            runner: "macos-latest"
           - backend: "SERIAL"
             cmake_build_type: "Debug"
+            cxx_standard: "17"
+            compiler: "c++"
+            runner: "macos-latest"
           - backend: "SERIAL"
             cmake_build_type: "Release"
+            cxx_standard: "17"
+            compiler: "c++"
+            runner: "macos-latest"
+          # Testing a compiler with mdspan implementation
+          - backend: "SERIAL"
+            cmake_build_type: "RelWithDebInfo"
+            cxx_standard: "23"
+            compiler: "$(brew --prefix llvm@18)/bin/clang++"
+            runner: "macos-15"
```

Comment:
Please spell out what this resolves to (what we tested before VS what we test now)

Reply:
I think you misunderstood my comment.
I was not saying that you should not use latest and you must spell out the actual versions, I was asking for a comment on GH highlighting differences in the CI builds with this PR.  I find it not always obvious how the matrix and include GH action directives play together.

Reply:
There are no "real" matrix builds here but only the 5 you see manually added.

Reply:
I found it clearer and (more maintainable) regardless by being explicit about the runners (and not auto-update them).

Reply:
So we are going from 4 to 5 builds, adding one C++23 build that is using LLVM Clang instead of AppleClang is that right?
Is it worth suffixing the name of that last build to highlight that difference?